src/telegrafs/actions/thunks.ts

```diff
@@ -168,8 +204,28 @@ export const removeTelegrafLabelsAsync = (
   labels: Label[]
 ): AppThunk<Promise<void>> => async (dispatch): Promise<void> => {
   try {
-    await client.telegrafConfigs.removeLabels(telegrafID, labels as ILabel[])
-    const telegraf = await client.telegrafConfigs.get(telegrafID)
+    if (labels.length === 0) {
+      throw new Error('No label found to delete')
+    }
+    const response = await deleteTelegrafsLabel({
+      telegrafID,
+      labelID: labels[0].id,
+    })
+
+    if (response.status !== 204) {
+      throw new Error(response.data.message)
+    }
+
+    const res = await apiGetTelegraf({
+      telegrafID,
+      headers: {Accept: 'application/json'},
```

Comment:
we should investigate why you need to do this vs why I'm not running into this same issue

Reply:
Does it has anything to do with an updated version of webpack dependencies? I haven't have updated it recently.

Reply:
It shouldn't have anything to do with that since it hinges upon the `generate` command. If you want to try merging in master and seeing if restarting the UI generates the expected functions without those `headers` being set

Reply:
I tried to merge in master, `yarn install`, and restarting UI on both Chrome and Firefox. I still have to set `headers` in order to get json objects. Very wired. We might want to revisit this in the future. But for now, we want to set the `headers` in order to receive json object.